api/v1beta1/flowcollector_types.go

```diff
@@ -358,6 +358,21 @@ type FlowCollectorFLP struct {
 	// kafkaConsumerBatchSize indicates to the broker the maximum batch size, in bytes, that the consumer will accept. Ignored when not using Kafka. Default: 10MB.
 	KafkaConsumerBatchSize int `json:"kafkaConsumerBatchSize"`
 
+	//+kubebuilder:default:={"flowLog"}
+	// outputRecordTypes is a list of record types to export from ["flowLog","newConnection","heartbeat","endConnection"]
+	// +optional
+	OutputRecordTypes *[]string `json:"outputRecordTypes,omitempty"`
```

Comment:
is there min number of elements in this list in another words can it be empty or it should have at minimum one entry ?

Reply:
That's a good question. I would like to gather some feedback here before taking a decision.

We could either keep this open using a string array or force specific cases using an enum.
The connection tracking stage is really flexible and will throw an error [only if a provided record type is unknown](https://github.com/netobserv/flowlogs-pipeline/blob/9b96359be572c2b9d0fcf5620d5be3386f8a955d/pkg/pipeline/extract/conntrack/conntrack.go#L240)
- Providing an empty array here could be an option to say ["only export prometheus metrics"](https://github.com/netobserv/network-observability-operator/pull/260). I probably need to update a bit the pipeline for that. 
 
Loki will become optionnal and the console plugin would be disabled.

- It would allow also to only export all / some connection events (`NewConnection` / `Heartbeat` / `EndConnection`) without flowlogs.
- Or only `flowLog` as you can see in the default value.

The console plugin UI adapts options accordingly in these cases.

What do you think ?

Reply:
IMO I think using enum with default here will be more appropriate but we should also hear what others thoughts on this

Reply:
> Providing an empty array here could be an option to say "only export prometheus metrics". I probably need to update a bit the pipeline for that.

IIUC, the proposal is that an empty array will be equivilant to an array with a single value "flowLog" which indicates disabling connection tracking?

Reply:
@ronensc no; I'm suggesting to be able to entirely skip Loki stage with empty array. Would it make sense ?

FYI I have played a bit with empty / default value and the yaml act as:
```
outputRecordTypes:
connectionUpdateInterval: 30s
connectionEndTimeout: 10s
```
-> is falling back to the default value `flowLog`

```
outputRecordTypes: []
connectionUpdateInterval: 30s
connectionEndTimeout: 10s
```
-> is explicitly using empty string but it feels a bit hacky

Reply:
Now I see. Hm... I can't think of a reason for users to set an empty array.
I mean, if the purpose is to skip Loki, wouldn't it be more expressive to remove Loki from the pipeline?
In addition, if the conntrack stage has no output records at all, then the prometheus stage which follows it won't be able to compute metrics.

Reply:
sure, let's move to an enum then :+1: 

thanks @msherif1234 & @ronensc